executor/aggregate.go

```diff
@@ -736,13 +743,13 @@ func (e *HashAggExec) waitPartialWorkerAndCloseOutputChs(waitGroup *sync.WaitGro
 	}
 }
 
-func (e *HashAggExec) waitFinalWorkerAndCloseFinalOutput(waitGroup *sync.WaitGroup) {
-	waitGroup.Wait()
-	close(e.finalOutputCh)
-}
-
 func (e *HashAggExec) prepare4ParallelExec(ctx context.Context) {
-	go e.fetchChildData(ctx)
+	fetchChildWorkerWaitGroup := &sync.WaitGroup{}
+	fetchChildWorkerWaitGroup.Add(1)
+	go func() {
+		e.fetchChildData(ctx)
+		fetchChildWorkerWaitGroup.Done()
```

Comment:
We can pass `fetchChildWorkerWaitGroup` as an input argument to `fetchChildData`.